File: 14.longest-common-prefix.js
/*
 * @lc app=leetcode id=14 lang=javascript
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
  if (strs.length === 0) return "";

  return strs.reduce((prefix, str) => {
    let i = 0;
    while (prefix[i] && str[i] && prefix[i] === str[i]) ++i;
    return prefix.slice(0, i);
  });
};
// @lc code=end

File: 14.longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python3
#
# [14] Longest Common Prefix
#

# Time:  O(n * k)
# Space: O(1)

from typing import List


# @lc code=start
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        for i in range(len(strs[0])):
            for j in range(1, len(strs)):
                if i >= len(strs[j]) or strs[j][i] != strs[0][i]:
                    return strs[0][:i]
        return strs[0]


# @lc code=end
